src/main/java/marquez/common/models/DatasourceType.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package marquez.common.models;
+
+public enum DatasourceType {
```

Comment:
not 100% sure this is related.. would like to consider `DatasourceType` separately from the changes to `DatasourceURN` and `DatasetURN`.

I'd like to eventually see #401 happen where the allowed types are config driven which will make it easier to extend Marquez Core simply with a config change. Enums can only be statically defined, so it may not support a config-driven list. No need to change now, but we should address when we tackle #401.

Reply:
Nice! I haven't given the _config driven_ approach to defining datasources much thought, but I see how it can be a powerful customization option. Thanks for starting the discussion! 🥇

Reply:
/cc @roaraya8 @ashulmanWeWork thoughts?

Reply:
Wanted to understand more - is it the case that we just won't be able to fully anticipate all the types of datasources that an organization using Marquez will want to track? I guess that would block an organization from using Marquez fully until the release where their datasource is an entry in the enum. If that's the case, why not just make this a free-form field instead of an enum? 

OTOH, suppose an admin wants to restrict the types, but only to the types that the admin deems appropriate? If so, seems healthier if that's established as a team decision and not a configuration owned by the Marquez admin. Or am I overthinking this? :)

Reply:
Let's move the discussion to the issue :)